### PR TITLE
[IMP] mail: unfollow from chatter without confirmation dialog

### PR DESCRIPTION
**Pull request by pka-odoo**
before this commit:
when user click on unfollow in chatter, it asks for confirmation to unfollow

after this commit:
when user click on unfollow in chatter, it will be unfollowed from particular
document straight away without any confirmation

Description of the issue/feature this PR addresses:
Do not show the confirmation popup while clicking on "Following" CTA in the chatter. 

Current behavior before PR:
the confirmation popup is showing while clicking on "Following" CTA in the chatter. 

Desired behavior after PR is merged:
when user click on unfollow in chatter, it will be unfollowed without any confirmation

Task:
https://www.odoo.com/web#id=1998188&action=327&model=project.task&view_type=form&menu_id=4720

Pad:
https://pad.odoo.com/p/r.b6ffbe3362a2d988a1989da79d03dcf6

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/static/src/js/followers.js [hunk 1]
```diff
@@ -239,30 +239,42 @@ var Followers = AbstractField.extend({
             })
             .then(this._reload.bind(this));
     },
+    /**
+     * Remove partners or channels from the followers
+     * @param {Array} [ids.partner_ids] the partner ids
+     * @param {Array} [ids.channel_ids] the channel ids
+     */
+    _messageUnsubscribe: function (ids) {
+        var self = this;
```
Comment (mge-odoo) [inline addons/mail/static/src/js/followers.js]: `self` is useless here ; you can replace all occurrences of `self` but `this` in this function?

File: addons/mail/static/src/js/followers.js [hunk 2]
```diff
@@ -239,30 +239,42 @@ var Followers = AbstractField.extend({
             })
             .then(this._reload.bind(this));
     },
+    /**
+     * Remove partners or channels from the followers
+     * @param {Array} [ids.partner_ids] the partner ids
+     * @param {Array} [ids.channel_ids] the channel ids
+     */
```
Comment (mge-odoo) [inline addons/mail/static/src/js/followers.js]: add `@private` in docstring

File: addons/mail/static/src/js/followers.js [hunk 3]
```diff
@@ -239,30 +239,42 @@ var Followers = AbstractField.extend({
             })
             .then(this._reload.bind(this));
     },
+    /**
+     * Remove partners or channels from the followers
```
Comment (mge-odoo) [inline addons/mail/static/src/js/followers.js]: you might want to specify `ids` in the docstring
```suggestion
     * Remove partners or channels from the followers
     *
     * @param {Object} ids
```

File: addons/mail/static/src/js/followers.js [hunk 4]
```diff
@@ -239,30 +239,42 @@ var Followers = AbstractField.extend({
...
+     * @param {Array} [ids.partner_ids] the partner ids
+     * @param {Array} [ids.channel_ids] the channel ids
+     */
+    _messageUnsubscribe: function (ids) {
+        var self = this;
+        var args = [
```
Comment (mge-odoo) [inline addons/mail/static/src/js/followers.js]: no need to create a variable for `args`, just put the dict in the `args` value

File: addons/mail/static/tests/chatter_tests.js [hunk 5]
```diff
@@ -2202,15 +2202,82 @@ QUnit.test('followers widget: follow/unfollow, edit subtypes', async function (a
...
+            }
+            return this._super.apply(this, arguments);
+        },
+        session: {partner_id: partnerID},
+    });
+
+    assert.strictEqual(form.$('.o_followers_count').text(), "2", 'should have 2 followers');
+    // click to unfollow current user
+    await testUtils.dom.click(form.$('.o_followers_follow_button'));
+    assert.strictEqual($('.modal').length, 0, 'a confirm modal should not be opened');
```
Comment (mge-odoo) [inline addons/mail/static/tests/chatter_tests.js]:
```suggestion
    assert.containsNone($, '.modal', 'a confirm modal should not be opened');
```

File: addons/mail/static/tests/chatter_tests.js [hunk 6]
```diff
@@ -2202,15 +2202,82 @@ QUnit.test('followers widget: follow/unfollow, edit subtypes', async function (a
...
+    assert.strictEqual(form.$('.o_followers_count').text(), "1", 'should have 1 follower');
+
+    await testUtils.dom.click(form.$('.o_followers_title_box [data-toggle="dropdown"]'));
+    // remove demo user from followers
+    await testUtils.dom.click(form.$('.o_remove_follower'));
+    assert.ok($('.modal').length, 'a confirm modal should be opened');
```
Comment (mge-odoo) [inline addons/mail/static/tests/chatter_tests.js]:
```suggestion
    assert.containsOnce($, '.modal', 'a confirm modal should be opened');
```

File: addons/mail/static/src/js/followers.js [hunk 7]
```diff
@@ -239,30 +239,42 @@ var Followers = AbstractField.extend({
...
+    },
     /**
      * Remove partners or channels from the followers
      * @param {Array} [ids.partner_ids] the partner ids
      * @param {Array} [ids.channel_ids] the channel ids
      */
     _unfollow: function (ids) {
         var self = this;
+        // do not prompt confirmation dialog on unsubscribe of current user.
+        if (_.contains(ids.partner_ids, this.partnerID)) {
```
Comment (mge-odoo) [inline addons/mail/static/src/js/followers.js]: here, you probably want to make the condition a little more precise, like:
```suggestion
        if (_.isEqual(ids.partner_ids, [this.partnerID]) && _.isEmpty(ids.channel_ids)) {
```

**Comment by pka-odoo** [comment]
Hello @mge-odoo Requested changes are done. Can you please check it now.
Thanks.

**Pull request closed without merging**